canal/canal_test.go

```diff
@@ -20,6 +20,7 @@ func Test(t *testing.T) {
 }
 
 type canalTestSuite struct {
+	h *testEventHandler
```

Comment:
do we need it in somewhere except for L67?

Reply:
Didnt fully cleanup the previously implemented test case, doing it now